hudi-common/src/main/java/org/apache/hudi/common/util/collection/ExternalSpillableMap.java

```diff
@@ -202,28 +202,20 @@ public R get(Object key) {
 
   @Override
   public R put(T key, R value) {
-    if (this.currentInMemoryMapSize >= maxInMemorySizeInBytes || inMemoryMap.size() % NUMBER_OF_RECORDS_TO_ESTIMATE_PAYLOAD_SIZE == 0) {
-      long tmpEstimatedPayloadSize = (long) (this.estimatedPayloadSize * 0.9
-          + (keySizeEstimator.sizeEstimate(key) + valueSizeEstimator.sizeEstimate(value)) * 0.1);
```

Comment:
This line is also a mistake, because it can trigger for the 1st record with `estimatedPayloadSize` as `0`, it then causes all the estimation of payload size far smaller than the real size.